lib/providers.go

```diff
@@ -62,16 +67,98 @@ func (m *MockDataSourceReader) UnmatchedCalls() []cty.Value {
 	return uc
 }
 
+// GetPluginFolder tries to compute the user plugin folder for the current user.
+// For windows: %APPDATA%/terraform.d/plugins
+// For linux: ~/terraform.d/plugins
+func GetPluginFolder() (string, error) {
+	homeDir, err := homedir.Dir()
+	if err != nil {
+		return "", err
+	}
+
+	if runtime.GOOS == "windows" {
+		return filepath.FromSlash(fmt.Sprintf("%s/AppData/Roaming/terraform.d/plugins", homeDir)), nil
+	}
+
+	return filepath.FromSlash(fmt.Sprintf("%s/.terraform.d/plugins", homeDir)), nil
+}
+
+// parseProviderValues retrives the values for hostname, namespace and provider name from the path.
+func parseProviderValues(provMeta discovery.PluginMeta) (*addrs.Provider, error) {
+	parts := strings.Split(filepath.ToSlash(provMeta.Path), "/")
+
+	partCount := len(parts)
+
+	// overall the path does not have to be that long for tf12
+	if partCount < 3 {
+		return nil, fmt.Errorf("No valid provider path: %s", provMeta.Path)
+	}
+
+	if parts[partCount-3] == "plugins" {
+		// plugins folder was initialized by terraform <=0.12
+		return &addrs.Provider{
+			Hostname:  addrs.DefaultRegistryHost,
+			//Namespace: addrs.LegacyProviderNamespace,
+			// Fake this, because terraform 13 will look for this namespace by default
+			Namespace: "hashicorp",
+			Type:      provMeta.Name,
+		}, nil
+	}
+
+	// for tf13 the path has to have a much longer length
+	if partCount < 6 {
+		return nil, fmt.Errorf("No valid provider path for tf13: %s", provMeta.Path)
+	}
+	
+	// this is for terraform 0.13 onwards
+	return &addrs.Provider{
+		Hostname:  svchost.Hostname(parts[partCount-6]),
+		Namespace: parts[partCount-5],
+		Type:      parts[partCount-4],
+	}, nil
+}
+
 // BuildProviderResolver returns a ProviderResolver able to find all providers
 // provided by plugins
 func BuildProviderResolver(dir string) (*ProviderResolver, error) {
-	var pluginDir = path.Join(dir, fmt.Sprintf(".terraform/plugins/%s_%s", runtime.GOOS, runtime.GOARCH))
 
-	pluginMetaSet := discovery.FindPlugins(plugin.ProviderPluginName, []string{pluginDir})
 	pluginsSchema := make(map[addrs.Provider]discovery.PluginMeta)
 
-	for k := range pluginMetaSet {
-		pluginsSchema[addrs.NewDefaultProvider(k.Name)] = k
+	// find plugins in project dir
+	projectPluginDir := path.Join(dir, ".terraform/plugins/")
+	osArch := fmt.Sprintf("%s_%s", runtime.GOOS, runtime.GOARCH)
+	// TODO: this check could probably be improved
+	_, err := os.Stat(path.Join(projectPluginDir, osArch))
+	isTf13 := os.IsNotExist(err)
```

Comment:
I wonder if we should add a cli flag so the user explicitely ask for a terraform 12 compatible behavior instead of having terraspec trying to guess in which condition it runs :thinking: 

Probably enough for now. I'll add a new flag in the future if it becomes too unpredictable